src/react/components/InterfaceEditor.tsx

```diff
@@ -0,0 +1,945 @@
+            )}
+          </Card.Body>
+        </Accordion.Collapse>
+      </Card>
+    </Accordion>
+  );
+};
+
+const getDefaultMapping = (params: {
+  interfaceType: AstarteInterface['type'];
+  interfaceAggregation: AstarteInterface['aggregation'];
+}): AstarteMapping => {
+  if (params.interfaceType === 'datastream' && params.interfaceAggregation === 'individual') {
+    return {
+      endpoint: '',
+      type: 'double',
+      explicitTimestamp: true,
+    };
+  }
+  return {
+    endpoint: '',
+    type: 'double',
+  };
+};
+
+interface MappingModalProps {
+  interfaceType: AstarteInterface['type'];
+  interfaceAggregation?: AstarteInterface['aggregation'];
+  mapping?: AstarteMapping;
+  onCancel: () => void;
+  onConfirm: (newMapping: AstarteMapping) => void;
+}
+
+const MappingModal = ({
+  interfaceType,
+  interfaceAggregation = 'individual',
+  mapping,
+  onCancel,
+  onConfirm,
+}: MappingModalProps): React.ReactElement => {
+  const [mappingDraft, setMappingDraft] = useState(
+    mapping || getDefaultMapping({ interfaceType, interfaceAggregation }),
+  );
+
+  const handleChange = useCallback((newMapping: AstarteMapping) => {
+    setMappingDraft(newMapping);
+  }, []);
+
+  const isValidMapping = AstarteMapping.validation.isValidSync(mappingDraft);
+
+  return (
+    <Modal show size="lg" centered onHide={onCancel}>
+      <Modal.Header closeButton>
+        <Modal.Title>{mapping != null ? 'Edit mapping' : 'Add new mapping'}</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <MappingEditor
+          interfaceType={interfaceType}
+          interfaceAggregation={interfaceAggregation}
+          mapping={mappingDraft}
+          onChange={handleChange}
+        />
+      </Modal.Body>
+      <Modal.Footer>
+        <Button variant="secondary" onClick={onCancel}>
+          Cancel
+        </Button>
+        <Button
+          variant="primary"
+          onClick={() => onConfirm(mappingDraft)}
+          disabled={!isValidMapping}
+        >
+          Confirm
+        </Button>
+      </Modal.Footer>
+    </Modal>
+  );
+};
+
+const computeInterfaceWarnings = (iface: AstarteInterface): { [key: string]: string } => {
+  const warnings: { [key: string]: string } = {};
+
+  if (iface.type === 'datastream' && iface.aggregation === 'object') {
+    const mappingsWarning =
+      'Interface mappings endpoints of depth 1 in Object aggregate interfaces are deprecated. The endpoint should have depth level of 2 or more (e.g. /my/endpoint).';
+    const endpoints = iface.mappings.map((mapping) => mapping.endpoint);
+    const endpointDepths = endpoints.map((endpoint) => endpoint.split('/').length - 1);
+    if (endpointDepths.some((depth) => depth < 2)) {
+      warnings.mappings = mappingsWarning;
+    }
+  }
+
+  const nameWarning =
+    'Interface name should be prefixed with a reverse domain name, and should use PascalCase (e.g. com.example.MyInterface)';
+  const nameGroups = iface.name.split('.').reverse();
```

Comment:
there is a specific regex to check fore checcking for warningse. You can copy it from the Elm side. We're using regex because we can copy-paste them directly from the astarte repo whenever they change.
for the warning you can use
```
const nameGroups = iface.name.split('.');
const lastGroup = nameGroups.pop();
if (!isPascalCase(lastGroup) || nameGroups.some(isPascalCase)) {
  return 'Interface name should..... ';
}
```

Reply:
I didn't find any regex for warnings. I found those for error validation which I already copied over to javascript.
However I agree this piece could be more readable, I'll give it another go to improve it

Reply:
I reworked those lines to be cleaner and use a simple regex to test for a good interface name